tests/components/rflink/test_init.py

```diff
@@ -196,6 +196,50 @@ async def test_send_command_invalid_arguments(hass, monkeypatch):
     assert not success, "send command should not succeed for unknown command"
 
 
+async def test_send_command_event_propagation(hass, monkeypatch):
+    """Test event propagation for send_command service."""
+    domain = "light"
+    config = {
+        "rflink": {"port": "/dev/ttyABC0"},
+        domain: {
+            "platform": "rflink",
+            "devices": {
+                "protocol_0_1": {"name": "test1"},
+            },
+        },
+    }
+
+    # setup mocking rflink module
+    _, _, protocol, _ = await mock_rflink(hass, config, domain, monkeypatch)
+
+    # default value = 'off'
+    assert hass.states.get(f"{domain}.test1").state == "off"
+
+    hass.async_create_task(
```

Comment:
Why create a task instead of awaiting the coroutine?

Reply:
It seems to be a common pattern in all the rflink tests, maybe it's a leftover when updating the tests from unittest to pytest?
@javicalle would you have time to clean that up?